tiledb/sm/rest/rest_profile.h

```diff
@@ -108,8 +108,22 @@ class RestProfile {
   /*              API               */
   /* ****************************** */
 
-  inline std::string name() {
-    return name_;
+  /**
+   * Returns the name of this profile.
+   *
+   * @return std::string The name of this profile.
+   */
+  inline const std::string name() const {
```

Comment:
These can avoid making string copies.
I don't have an opinion between returning `const std::string &` versus `std::string_view` but `const std::string` shouldn't be the return value